source/h1_encoder.c

```diff
@@ -691,17 +720,13 @@ int aws_chunk_line_from_options(struct aws_http1_chunk_options *options, struct
     if (AWS_OP_SUCCESS != aws_byte_buf_init(chunk_line, options->chunk_data->allocator, chunk_line_size)) {
         return AWS_OP_ERR;
     }
-    if (!s_populate_chunk_line_buffer(chunk_line, options)) {
-        /* The caller will free the chunk, which will free the chunk line in the process */
-        return AWS_OP_ERR;
-    }
+    AWS_ASSERT(s_populate_chunk_line_buffer(chunk_line, options));
```

Comment:
This whole thing will get compiled out in release. You'll want to do one of these annoying things:
```
bool populate_chunk_line_success = s_populate_chunk_line_buffer(chunk_line, options);
AWS_ASSERT(populate_chunk_line_success);
(void)populate_chunk_line_success; <-- avoid compiler warnings about unused variables
```
or
```
if (!s_populate_chunk_line_buffer(chunk_line, options)) {
    AWS_ASSERT(0);
}
```

Reply:
done.